### PR TITLE
feat(mp-8rc9): operator-run league tie-breakers + league/pool terminology cleanup

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

Operator-driven league tie-breakers (kendo team daihyosen tie-breaks) plus a league/pool terminology cleanup. Three phases, plus a tri-review hardening pass.

- **Terminology:** `format: league` competitions now read "League table" / "League tie-breaker" everywhere â no "Pool A" vocabulary leaks into the operator, standings, or schedule surfaces (display-layer only; the internal single-pool data is unchanged).
- **Config validation:** league (and playoffs) reject/zero the meaningless `poolSize`/`poolWinners` settings server-side; swiss is left untouched.
- **Operator-driven tie-breaks:** team leagues no longer auto-inject a daihyosen. The engine detects *consequential* ties (configurable `leagueTiebreakTopN` 3/4, honouring the two-joint-3rd-places convention) and blocks completion (`AutoCompleteAwaitingLeagueTiebreak`) until the operator decides. A new amber banner lets the operator **run a tie-breaker**, **remove** an unscored one, or **accept shared ranks**. 3+-way ties run a round-robin; a rock-paper-scissors cycle blocks for a manual override.
- **Terminology + tests:** the feature is named "tie-breaker" (daihyosen) throughout — user-facing copy, Go symbols, routes (`/league-tiebreak/*`), config keys, CSS, and filenames (the knockout "playoffs" format is untouched). Added edge-case tests for the consequential-band boundaries, the two-joint-3rd exemption, default Top-N, whole-band ties, below-band shared ranks, and non-team/non-league.

## Why

A league is a single round-robin table, not a set of pools â but league-format team comps were rendering "Pool A", accepting pool/playoff settings, and naming the DH match `Pool A-DH-0`. FIK does not specify league tie-break procedure, so (per product owner) the daihyosen is operator-driven, not automatic. The engine's end-of-league tieâDH trigger was already correct; the work was terminology, config validation, the configurable consequential-position rules, and the operator surface.

## Files changed

| File | What |
|---|---|
| `internal/state/models.go` | New `LeagueTiebreakTopN` / `LeagueTwoThirdPlaces` / `LeagueTiebreakFinalized` config fields |
| `internal/engine/league_tiebreak.go` | `LeagueTiebreakCandidates`, `GenerateLeagueTiebreakMatches`, consequential-tie logic |
| `internal/engine/competition.go` | `AutoCompleteAwaitingLeagueTiebreak` outcome + per-group completion gate; `leagueGroupHasDH` |
| `internal/engine/pool_daihyosen_test.go` | Tests updated for team-league no longer auto-injecting |
| `internal/engine/league_tiebreak_test.go`, `league_tiebreak_multigroup_test.go` | Engine tests incl. multi-group + no-wedge regression |
| `internal/mobileapp/handlers_league_tiebreak.go` | GET candidates (public) + POST/DELETE/finalize (admin), with validation guards |
| `internal/mobileapp/server.go` | Wire public read + admin mutation handler groups |
| `internal/mobileapp/handlers_competition.go` | Normalize pool config; lock league tie-breaker config post-start |
| `internal/mobileapp/handlers_match.go` | Broadcast on the awaiting-playoff outcome |
| `internal/mobileapp/handlers_*_test.go` | Handler tests (auth split, dup names, partial-group, running guard, immutability) |
| `web-mobile/js/admin_pools.jsx` | Operator tie-breaker banner (amber `.league-tiebreak`) |
| `web-mobile/js/admin_competition_settings.jsx` | League tie-breaker settings (Top 3/4 + two joint 3rds) |
| `web-mobile/js/viewer_utils.jsx` | Shared `leagueAwareLabel` helper (deduped across 5 files) |
| `web-mobile/js/admin_scoring_individual.jsx`, `admin_scoring_team.jsx`, `admin_shiaijo.jsx`, `viewer_standings.jsx` | League-aware labels |
| `web-mobile/js/api_client.jsx` | Four league-tiebreak client methods |
| `web-mobile/css/styles.css` | `.league-tiebreak` amber banner block (tokenized) |
| `specs/openapi.yaml` | Documented the four league-tiebreak endpoints |
| `web-mobile/vitest.setup.render.js` | Provide `window.poolLabel`/`leagueAwareLabel` to mounted render tests |

## Screenshots

**Operator tie-breaker banner** (amber "attention" treatment â distinct from the navy running-state and red danger; "Run tie-breaker" is the primary action):

League tie-breaker banner

**League tie-breaker settings** (configurable consequential positions + the two-joint-3rd-places convention):

League tie-breaker settings

## Test plan

- [x] `make go/test` passes (lint + security scan + tests) â engine 85.6%, mobileapp 85.6%, all packages â¥85%
- [x] New/updated unit tests cover the change â engine multi-group + no-wedge regression, handler auth-split / dup-names / partial-group / running-guard, config-immutability, render-smoke
- [x] Manual browser verification â drove a 4-team all-draw league end-to-end: banner â Run tie-breaker (generates the 6-bout round-robin) â Remove â Accept shared ranks â status `completed`; verified Top 3/4 + two-thirds settings persist; confirmed "League table" labels and no "Pool" vocabulary
- [x] Screenshots added above
- [x] No new console errors or warnings

Closes mp-8rc9

ð¤ Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/mobileapp/handlers_competition.go [hunk 1]
```diff
@@ -180,6 +198,24 @@ func validateSwissConfig(comp *state.Competition) error {
...
+	}
+	switch comp.LeaguePlayoffTopN {
+	case 0, 3, 4:
+		// 0 = unset (will default to 3 at draw time); 3 and 4 are the only
+		// legal explicit values (per owner decision Q1).
+	default:
+		return fmt.Errorf("leaguePlayoffTopN must be 3 or 4 (got %d)", comp.LeaguePlayoffTopN)
+	}
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_competition.go]: `ValidateCompetitionTeamSize` (called on every create and edit) enforces `Kind == "team" ⟺ TeamSize >= 2`, making `Kind == "team"` the canonical and sufficient gate. No competition can have `Kind == ""` with `TeamSize > 0` via this API — the previous defensive `TeamSize > 0` check was reverted based on earlier review feedback for exactly this reason.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/state/models.go [hunk 2]
```diff
@@ -345,6 +345,48 @@ type Competition struct {
...
+	// an operator-initiated play-off when teams are tied at the end of league
+	// play. Meaningful only for team-league competitions (Format == "league" &&
+	// TeamSize > 0). Allowed values are 3 (default) and 4.
+	//
+	// A "consequential" tied group is one whose position range intersects the
+	// band [1..LeaguePlayoffTopN]. For example, with TopN=3 a tie involving
+	// teams at positions 1–2 is consequential (it affects who wins or is
+	// runner-up), but a tie at positions 4–5 is not. When all teams in a tied
+	// group occupy positions strictly BELOW TopN (i.e. every position ≥ TopN)
+	// the group is non-consequential and no play-off is needed.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/state/models.go]: Fixed: the `LeagueTiebreakTopN` doc comment now accurately describes the band as positions 1..TopN inclusive. A tie group is consequential when its position range intersects [1..TopN]. The ambiguous "strictly BELOW TopN" phrasing is gone.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/mobileapp/handlers_league_tiebreak.go [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+			if !dhIDs[m.ID] {
+				filtered = append(filtered, m)
+			}
+		}
+
+		// Persist the filtered pool-match list. Race risk: a concurrent score
+		// write could land between our Load and this Save, but the delete is
+		// an operator-explicit action and we already guard against scored
+		// matches above, so a last-write-wins outcome is acceptable here.
+		// A future refactor can wrap this in WithTransaction for stronger
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_league_tiebreak.go]: Fixed: the DELETE handler now wraps its load→guard→filter→save inside `store.WithTransaction`, holding the per-comp lock across the entire read-modify-write. A concurrent score write can no longer land between the load and the save.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/state/models.go [hunk 4]
```diff
@@ -345,6 +345,48 @@ type Competition struct {
...
+	// an operator-initiated tie-breaker when teams are tied at the end of league
+	// play. Meaningful only for team-league competitions (Format == "league" &&
+	// TeamSize > 0). Allowed values are 3 (default) and 4.
+	//
+	// A "consequential" tied group is one whose position range intersects the
+	// band [1..LeagueTiebreakTopN]. For example, with TopN=3 a tie involving
+	// teams at positions 1–2 is consequential (it affects who wins or is
+	// runner-up), but a tie at positions 4–5 is not. When all teams in a tied
+	// group occupy positions strictly BELOW TopN (i.e. every position ≥ TopN)
+	// the group is non-consequential and no tie-breaker is needed.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/state/models.go]: Fixed: the doc comment for `LeagueTiebreakTopN` no longer uses "strictly BELOW TopN". It now reads "a tie group is consequential when its position range intersects [1..TopN]" — unambiguous and consistent with the engine implementation.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/league_tiebreak.go [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+	// (Phase 3b finalize endpoint), treat the competition as having no
+	// consequential ties so MaybeAutoCompletePools can transition normally.
+	if comp.LeagueTiebreakFinalized {
+		return nil, nil
+	}
+
+	standings, err := e.CalculatePoolStandings(compID)
+	if err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_tiebreak.go:135]: Fixed: `LeagueTiebreakCandidates` now returns an empty slice while any regular (non-DH) pool match is still scheduled or in-progress, so the banner cannot appear prematurely. `TestLeagueTiebreakCandidates_EmptyUntilRegularComplete` pins this: three round-robin matches, one completed → empty candidates; all three completed as draws → candidates appear.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/league_tiebreak.go [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+		poolName = pn
+		for _, s := range ps {
+			if nameSet[s.Player.Name] {
+				tiedGroup = append(tiedGroup, s)
+			}
+		}
+	}
+	if len(tiedGroup) == 0 {
+		return nil, validationErrorf("none of the requested teams found in standings for competition %s", compID)
+	}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/league_tiebreak.go]: Fixed: `GenerateLeagueTiebreakMatches` now rejects unknown team names (with a `ValidationError`) and duplicate names at the engine level, independent of any handler gate. `TestGenerateLeagueTiebreakMatches_RejectsBadNames` covers both the unknown-name and duplicate-name cases.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/mobileapp/handlers_league_tiebreak.go [hunk 7]
```diff
@@ -0,0 +1,428 @@
...
+		if hadDup {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "teamNames contains duplicate entries"})
+			return
+		}
+
+		allMatches, err := store.LoadPoolMatches(id)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_league_tiebreak.go]: Fixed: the DELETE handler now loads the competition at the top of `WithTransaction` and returns 400 with a clear message if `Format != CompFormatLeague || Kind != "team"`. `TestLeagueTiebreakDelete_RejectsNonTeamLeague` pins this — a mixed-comp DELETE gets 400 and leaves the existing DH matches untouched.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/mobileapp/handlers_match.go [hunk 8]
```diff
@@ -226,6 +226,13 @@ func tryAutoCompletePools(c *gin.Context, eng ScoringEngine, hub Broadcaster, co
 		// matches may now be live — refresh without a full status change.
 		hub.Broadcast(EventMatchUpdated, gin.H{"competitionId": compID})
 		hub.Broadcast(EventScheduleUpdated, nil)
+	case engine.AutoCompleteAwaitingLeagueTiebreak:
+		// All regular team-league pool matches are complete but consequential ties
+		// remain — the operator must either generate tie-breaker matches or finalize
+		// shared ranks via the league-tiebreak endpoints (Phase 3b). Signal a
+		// match_updated event so connected clients reload standings and display
+		// the awaiting-playoff banner.
+		hub.Broadcast(EventMatchUpdated, gin.H{"competitionId": compID})
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_match.go:235]: Fixed: the comment in `tryAutoCompletePools` for the `AutoCompleteAwaitingLeagueTiebreak` case now says "tie-breaker" (not "playoff"). The `AutoCompleteAwaitingLeagueTiebreak` doc comment in `competition.go` was also updated to document both required broadcasts (`EventMatchUpdated` + `EventScheduleUpdated`).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/mobileapp/handlers_league_tiebreak.go [hunk 9]
```diff
@@ -0,0 +1,474 @@
...
+		candidates, err := eng.LeagueTiebreakCandidates(id)
+		if err != nil {
+			var notFound *engine.NotFoundError
+			if errors.As(err, &notFound) {
+				c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+				return
+			}
+			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+			return
+		}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_league_tiebreak.go:216]: Fixed: the POST handler now loads the competition immediately after body binding, returns 404 if not found, and returns 400 with a clear "league tie-breaker endpoints apply only to team-league competitions" message for non-team-league comps — before ever calling `LeagueTiebreakCandidates`. All 5 `err.Error()` raw-error leakage instances in this file were swept and replaced with the opaque-500 pattern (`log.Printf` + `"internal error"`). Class swept, no other occurrences in this diff. `TestLeagueTiebreakPost_RejectsNonTeamLeague` pins both cases (mixed-format and individual-kind).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/engine/competition.go [hunk 10]
```diff
@@ -51,13 +51,31 @@ const (
...
 	AutoCompletePoolsResolved AutoCompleteOutcome = 4
+	// AutoCompleteAwaitingLeagueTiebreak means all regular team-league matches are
+	// done and there is at least one consequential tie (a group of tied teams whose
+	// position range intersects [1..LeagueTiebreakTopN], adjusted for the two-joint-
+	// 3rd-places convention). The competition stays in CompStatusPools; the engine
+	// did NOT auto-inject any DH matches. The operator must decide which teams play
+	// a supplementary tie-breaker (Phase 3b will add the operator endpoint). Until the
+	// operator acts (or accepts shared ranks), the competition cannot transition to
+	// CompStatusComplete. Callers should broadcast EventScheduleUpdated so the UI
+	// can show the "awaiting tie-breaker decision" banner.
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/competition.go]: Fixed both the comment and the caller: `tryAutoCompletePools` in `handlers_match.go` now broadcasts `EventMatchUpdated` (reload standings) AND `EventScheduleUpdated` (show tie-breaker banner) for this outcome. The `AutoCompleteAwaitingLeagueTiebreak` doc comment in `competition.go` was updated to document both required broadcasts and remove the stale "Phase 3b will add the operator endpoint" note.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: web-mobile/js/admin_pools.jsx [hunk 11]
```diff
@@ -241,6 +250,168 @@ function AdminPools({ c, pools, poolMatches, standings, tweaks, onEditScore, pas
...
+              ) : (
+                <button
+                  type="button"
+                  className="btn btn--sm btn--danger btn--ghost"
+                  disabled={tiebreakActionBusy}
+                  onClick={() => handleTiebreakRemove(names)}
+                >
+                  {tiebreakBusyAction === removeKey && <span className="spinner" />}
+                  Remove unscored tie-breaker
+                </button>
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_pools.jsx]: Fixed in 6a16be46. Added `dhMatchScoredForGroup()` in `admin_pools.jsx` — it returns `true` when any DH match for the group has `status === "running"`, `status === "completed"`, or a non-empty `winner` (the exact conditions the DELETE handler checks). The "Remove unscored tie-breaker" button now receives `disabled={tiebreakActionBusy || dhScored}` and a `field__hint` explains why it is unavailable when `dhScored` is true. Class swept: the only other DH match consumption is the read-only score display, which needs no guard.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: internal/mobileapp/handlers_league_tiebreak.go [hunk 12]
```diff
@@ -0,0 +1,498 @@
...
+				return nil, nil
+			}
+			comp.LeagueTiebreakFinalized = true
+			return comp, nil
+		})
+		if err != nil {
+			log.Printf("league-tiebreak finalize UpdateCompetitionChanged(%s): %v", id, err)
+			c.JSON(http.StatusInternalServerError, gin.H{"error": "internal error"})
+			return
+		}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_league_tiebreak.go:470]: Fixed in 6a16be46. Added a `notTeamLeague bool` sentinel alongside `notFoundFlag` / `alreadyComplete` in the finalize handler. Inside `UpdateCompetitionChanged`, after the nil-comp check and before the `alreadyComplete` check, the guard `if comp.Format != state.CompFormatLeague || comp.Kind != "team"` sets the sentinel and returns `nil, nil`. After the call, a `notTeamLeague` branch returns 400 with the same error string used by POST and DELETE. Added `TestLeagueTiebreakFinalize_RejectsNonTeamLeague` covering both mixed-format and individual-league cases, matching the pattern of `TestLeagueTiebreakPost_RejectsNonTeamLeague`. Class swept: POST and DELETE already had guards; finalize was the only missing sibling.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
